rdrf/rdrf/admin.py

```diff
@@ -178,6 +183,93 @@ def get_readonly_fields(self, request, obj=None):
         return () if obj is None else ("code",)
 
 
+class ActivationKeyExpirationListFilter(admin.SimpleListFilter):
+    title = _('activation key expired')
+    parameter_name = 'activation_key_expired'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('True', _('True')),
+            ('False', _('False')),
+        ]
+
+    def queryset(self, request, queryset):
+        expired_emails = []
+        unexpired_emails = []
+        for profile in queryset:
```

Comment:
Should be chaining ORM queries to the existing queryset rather than using a `for` loop.

Reply:
As discussed, my suggestion to chain ORM queries doesn't work here because you need to call a function on the profile `activation_key_expired()`. 

You  could look at using List Comprehensions to reduce code.

Something like:
```
expired_profiles = [profile.id for profile in queryset if profile.activation_key_expired()]
        
queryset.objects.filter(id__in=[expired_profiles]) 
queryset.objects.exclude(id__in=[expired_profiles])
```


https://realpython.com/list-comprehension-python/#using-list-comprehensions
https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions